junit5/src/main/java/com/linecorp/armeria/testing/junit5/server/ServiceRequestContextCaptor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.server;
+
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.function.Function;
+
+/**
+ * Captures the {@link ServiceRequestContext}s.
+ */
+public final class ServiceRequestContextCaptor {
```

Comment:
I think we need to provide this class to JUnit4 users as well. So how about:
- Moving this class to `com.linecorp.armeria.testing.server` in the same module (JUnit5)
- Copying the class to the JUnit4 module using Gradle script?

Reply:
Yes, I'm also wondering whether we should replicate this or using Gradle script to copy

Reply:
IIUC, We can't clear the captured contexts between test methods when a user used `@ClassRule`, so I was rather thinking not supporting this feature for JUnit 4.

Reply:
Can't we do:
```java
private static final ServiceRequestContextCaptor captor = new ServiceRequestContextCaptor();

@ClassRule
public static final ServerRule server = new ServerRule() {
    @Override
    protected void configure(ServerBuilder sb) throws Exception {
        sb.decorator(captor.decorator());
        ...
    }
}

@Before
public void setUp() {
    captor.clear();
}
```
I thought exposing `ServiceRequestContextCaptor.decorator()` was for allowing a user to explicitly use the captor. 🤔

Reply:
Ah I got your point now. Yeah, a user can do that. I was rather rejecting the idea of making `ServerRule` has the same feature set with `ServerExtension`, which is not possible obviously. I agree that a user can create a captor manually and use it. +1 for:

> - Moving this class to `com.linecorp.armeria.testing.server` in the same module (JUnit5)
> - Copying the class to the JUnit4 module using Gradle script?